logstash-plugins/usagetype-plugin/src/main/java/org/reactome/logstash/plugins/usagetype/util/Range.java

```diff
@@ -0,0 +1,101 @@
+package org.reactome.logstash.plugins.usagetype.util;
+
+/**
+ * Represents a range of comparable numerics. This is intended for NON-overlapping ranges ONLY.
+ * @author sshorser
+ *
+ * @param <T>
+ */
+public class Range<T extends Number & Comparable<T>> implements Comparable<Range<T>>
+{
+	private T start;
+	private T end;
+
+	public Range(T start, T end)
+	{
+		if (start.compareTo(end) > 0)
+		{
+			throw new IllegalArgumentException("Start must be less than end. Start: " + start.toString() + "; End: " + end.toString());
+		}
+		this.start = start;
+		this.end = end;
+	}
+
+	/**
+	 * Checks a value and indicates if it is in this range, or outside this range.
+	 * @param value - the value to check.
+	 * @return 0 if value is within the bounds of this range. -1 if value is less than this range's start. +1 if value is greater than this range's end.
+	 */
+	public int isInRange(T value)
+	{
+		if (value.compareTo(start) >= 0 && value.compareTo(end) <= 0)
+		{
+			return 0;
+		}
+		else if (value.compareTo(start) < 0)
+		{
+			return -1;
+		}
+		else //if (value.compareTo(end) > 0)
```

Comment:
Just checking - did you want to leave the commented out if as a description of the condition for when the else block is run?

Reply:
That was probably intentional, but it's been quite a while since I've looked at this, so I will need to have another look.